v2/index/generator.go

```diff
@@ -60,6 +60,17 @@ func Generate(v1 io.ReadSeeker) (Index, error) {
 			return nil, err
 		}
 
+		// Ignore frames whose length is zero.
+		// They don't contain a CID nor block, so they're not useful.
+		// Plus, some existing CARv1 users append null paddings to files,
+		// so it's helpful if we skip them here too.
+		// TODO: In the future, the CARv1 spec should allow us to skip
+		// the rest of the input here. If not, we should do something
+		// more performant here, such as reading zeros in chunks.
+		if length == 0 {
+			continue
```

Comment:
As mentioned in our meeting: I would *vastly* prefer this to be a `break`

Unless I am misunderstanding the codeflow...

Reply:
See the TODO :) That's pending an addition to the CARv1 spec to explicitly say that treating a null byte as an EOF is okay.

If we all agree that will definitely go into the carv1 spec, then I can make the change here, but I'm hesitant to make v2.0.0 behave in a way that's not following the spec.

Reply:
Then again, it seems like we all agree the spec should be changed that way, so I'll go ahead and change it for a break.

@rvagg please say if you disagree

Reply:
As per https://github.com/ipld/go-car/issues/140#issuecomment-874437177 I think we could make it a CARv1 reader thing, but as an opt-in when opening a CAR—signal that a `0==EOF`, otherwise `0==err`. It's not great to have Filecoin continuing to use a fork and if length data has been discarded then there's not much we can do (I did raise this very early on and iirc I was told that length data persisted through the deal!).

Reply:
Agreed. I think we need to flex the spec to not break v1 considering existing usage pattern; there is no getting away from that.